### PR TITLE
Deploy agent with hostNetwork on k8s

### PR DESCRIPTION
**Pull request by ChrsMark**
We need to deploy Agent using `hostNetwork` option so as to have access to node's ports directly for testing endpoints like these of scheduler's and controllermanager.

### CODE REVIEW

**Review by mtojek — APPROVED** [review]
WFG

**Comment by mtojek** [comment]
BTW To verify features introduced in PRs you can use the `test/packages` stuff to quickly write up samples. In this case you could verify the connectivity (not required, just a hint).

File: internal/testrunner/runners/system/servicedeployer/kubernetes.go [hunk 1]
```diff
@@ -95,7 +95,7 @@ func (ksd KubernetesServiceDeployer) SetUp(ctxt ServiceContext) (DeployedService
 
 	ctxt.Name = kind.ControlPlaneContainerName
 	ctxt.Hostname = kind.ControlPlaneContainerName
-	ctxt.Agent.Host.NamePrefix = "kind-fleet-agent-"
+	ctxt.Agent.Host.NamePrefix = "kind-control-plane"
```
Comment (mtojek) [inline internal/testrunner/runners/system/servicedeployer/kubernetes.go:98]: nit: missing trailing dash or is it on purpose?

I'm not sure about implications of this change. Would you mind sharing more context? Will it affect any existing integrations

Comment (mtojek) [inline internal/testrunner/runners/system/servicedeployer/kubernetes.go:98]: I looked into the kubectl dump for the failed build: https://beats-ci.elastic.co/job/Ingest-manager/job/elastic-package/job/PR-332/1/artifact/build/kubectl-dump.txt

and I see that there is a mix of `kind-control-plane` and `kind-fleet-agent-clusterscope`. I admit I'm bit lost here.

**Review by mtojek — CHANGES_REQUESTED** [review]
I will block this PR from merging, so that we're sure that we won't merge anything breaking here.

**Comment by ChrsMark** [comment]
Deploying Agent using `hostNetwork` will result into a different hostname. Pod will fallback to node's name for the hostname. This will lead into having Agent registered as `kind-control-plane` (name of the host) in fleet server while the test will be waiting for `kind-fleet-agent-*` and it will stall there forever (==timeout).

**Comment by mtojek** [comment]
Would you mind adding a comment in code around NodePrefix, justifying the "kind-control-plane"? I'm a bit cautious while dealing with host network and potential implications, but if there is no other option, let's do this. I undestand that it will only impact the kubernetes integration. Do you think that it won't be a problem in the future? Let's say, we would like to build the integration with istio.

**Review by mtojek — APPROVED** [review]
Not blocking, thank you for the explanation.

**Comment by ChrsMark** [comment]
> Would you mind adding a comment in code around NodePrefix, justifying the "kind-control-plane"? I'm a bit cautious while dealing with host network and potential implications, but if there is no other option, let's do this. I undestand that it will only impact the kubernetes integration. Do you think that it won't be a problem in the future? Let's say, we would like to build the integration with istio.

We use `hostNetwork` in our actual manifests too so it shouldn't be a problem for this testing scenario since it is actually identical to the real world's case:
https://github.com/elastic/beats/blob/188345dc96876011fd397ad7fd2f721459e102d7/deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml#L457
https://github.com/elastic/beats/blob/188345dc96876011fd397ad7fd2f721459e102d7/deploy/kubernetes/metricbeat-kubernetes.yaml#L146

I cannot think of any specific issue we can hit in the future because of this. Other k8s services/apps will be available using DNS resolution of k8s (using k8s service's name like what we do for kube-state-metrics). `hostNetwork` is also useful for autodiscovery (out of package's scope) so as to enable a k8s api watcher on node's level. 

In general, I think we are on the safe side with this ("hold my beer" 😄 ).

**Comment by mtojek** [comment]
@ChrsMark I think this one is good to go now.

**Pull request merged**